core/src/main/scala/core/database/QueryRunner.scala

```diff
@@ -4,26 +4,36 @@ import javax.sql.DataSource
 
 import scala.concurrent.{ExecutionContext, Future}
 
-class QueryRunner(
-    db: Database,
-    ec: ExecutionContext
-) {
-  def run[A](query: Query[A]): Future[A] =
+/**
+  * A class who can run a Query.
+  */
+sealed trait QueryRunner {
+  val ds: DataSource
+
+  val ec: ExecutionContext
+
+  val db: Database = Database(ds)
+
+  def apply[T](query: Query[T]): Future[T]
+}
+
+final class SimpleRunner(val ds: DataSource, val ec: ExecutionContext)
```

Comment:
```scala
trait WithConnection {
  def apply[T](f: Connection => T): T
}
```

... instances for such `WithConnection` can easily be provided in "2 flavours" in a Play app using `Database`.

```scala
class WithPlayTransaction(db: Database) extends WithConnection {
  def apply[T](f: Connection=> T): T = db.withTransaction(f)
}
```

Then core can provide a single runner factory.

```scala
object QueryRunner {
  private class Default(wc: WithConnection)(implicit ec: ExecutionContext) extends QueryRunner {
    def apply[T](q: Query[T]): Future[T] =
      Future(wc(query.f))
  }

  // Default factory
  def apply(wc: WithTransaction)(implicit ec: ExecutionContext): QueryRunner = new Default(wc)
}
```

Usage:

```scala
def inPlay[T](db: Database, q: Query[T])(implicit ec: ExecutionContext): Future[T] = {
  val runner = QueryRunner(new WithPlayTransaction(db))
  runner(q)
}
```